### DISCUSSION TITLE
Proxy Auth

### DISCUSSION THREAD

**Discussion by DarksideVT** [CrowdStrike/falconpy#1125] [Q&A] [answered]
I have a proxy that needs authentication in the form of `username:password@proxyurl.com:8080`. I'm finding that it doesn't seem to like that format when I pass it as a proxy dict to OAuth2 and I would assume the same could be said about any other Class. Am I missing something?

**Comment by DarksideVT** [comment]
A little bit more info is that I can use `curl -x "username:password@proxy.com:8080" -X POST "https://api.us-2.crowdstrike.com/OAuth2/token" -H "accept...` and get through no problem.

**Comment by jshcodes** [comment, marked as answer]
Hi @DarksideVT -

Thank you for the question!

Requests handles proxy authorization formatting a little strangely. Did some digging, and this appears to work for me in my testing environment.  Give this syntax a shot and let us know.

```python
import logging
from falconpy import Hosts

logging.basicConfig(level=logging.DEBUG)

proxies = {
    "https": "https://PROXYUSER:PROXYPASSWORD@PROXYHOST:PROXYPORT"
}
hosts = Hosts(debug=True, proxy=proxies)

hosts.query_devices_by_filter_scroll()
```

> We will make sure this gets added to the documentation. Thank you!
